test/framework/src/main/java/org/elasticsearch/test/rest/yaml/restspec/ClientYamlSuiteRestApiParser.java

```diff
@@ -146,4 +147,20 @@ public ClientYamlSuiteRestApi parse(String location, XContentParser parser) thro
 
         return restApi;
     }
+
+    private static class Parameter {
+        private boolean required;
+        public boolean isRequired() {
+            return required;
+        }
+        public void setRequired(boolean required) {
+            this.required = required;
+        }
+    }
+
+    private static boolean getRequired(XContentParser parser) throws IOException {
+        ObjectParser<Parameter, Void> objParser = new ObjectParser<>("", true, Parameter::new);
```

Comment:
Ah! More like:
```
private static final ObjectParser<Parameter, Void> PARAMETER_PARSER = new ObjectParser<>("parameter", true, Parameter:true);
static {
  PARAMETER_PARSER.declareBoolean(Parameter::setRequired, new ParseField("required");
}
```

And they you can call it like `PARAMETER_PARSER.parse(parser, null).isRequired()` like you have. That way we reuse the parser. It probably isn't a big deal, but it is what we do everywhere else so we may as well do it here.